cornflow-dags/DAG/two_dimension_bin_packing/solvers/right_corner_model.py

```diff
@@ -0,0 +1,295 @@
+            )
+
+        model.final_item_height_a = Constraint(model.Items, rule=final_item_height_a)
+
+        def final_item_height_b(model, item):
+            return model.FinalItemHeight[item] - model.ItemHeight[
+                item
+            ] <= model.BigM * (1 - model.ItemIn[item])
+
+        model.final_item_height_b = Constraint(model.Items, rule=final_item_height_b)
+
+        def final_width_domain(model, item):
+            return model.FinalItemWidth[item] <= model.BigM * model.ItemIn[item]
+
+        model.final_width_domain = Constraint(model.Items, rule=final_width_domain)
+
+        def final_height_domain(model, item):
+            return model.FinalItemHeight[item] <= model.BigM * model.ItemIn[item]
+
+        model.final_height_domain = Constraint(model.Items, rule=final_height_domain)
+
+        def no_width_intercept(model, item_1, item_2):
+            if item_1 != item_2:
+                return (
+                    model.XCoordinate[item_1]
+                    + model.FinalItemWidth[item_2]
+                    - model.XCoordinate[item_2]
+                    <= model.BigM * model.XInterference[item_1, item_2]
+                )
+            else:
+                return Constraint.Skip
+
+        model.no_width_intercept = Constraint(
+            model.Items, model.Items, rule=no_width_intercept
+        )
+
+        def no_height_intercept(model, item_1, item_2):
+            if item_1 != item_2:
+                return (
+                    model.YCoordinate[item_1]
+                    + model.FinalItemHeight[item_2]
+                    - model.YCoordinate[item_2]
+                    <= model.BigM * model.YInterference[item_1, item_2]
+                )
+            else:
+                return Constraint.Skip
+
+        model.no_height_intercept = Constraint(
+            model.Items, model.Items, rule=no_height_intercept
+        )
+
+        def no_interference(model, item_1, item_2):
+            if item_1 != item_2:
+                return (
+                    model.XInterference[item_1, item_2]
+                    + model.XInterference[item_2, item_1]
+                    + model.YInterference[item_1, item_2]
+                    + model.YInterference[item_2, item_1]
+                    <= 3
+                )
+            else:
+                return Constraint.Skip
+
+        model.no_interference = Constraint(
+            model.Items, model.Items, rule=no_interference
+        )
+
+        # objective function
+        def objective_function(model):
+            return sum(
+                model.ItemIn[item] * model.ItemValue[item] for item in model.Items
+            )
+
+        model.obj = Objective(rule=objective_function, sense=maximize)
+
+        return model
+
+    def solve(self, options):
+        data = self.get_pyomo_dict_data()
+        model = self.get_model()
+        model_instance = model.create_instance(data=data)
+        solver_name = options.get("solver", "cbc")
```

Comment:
@hugolarzabal For the solvers that use a mip model, the solver option generally looks like "right_corner.cbc" or "right_corner.gurobi" (If you look to the config schema, you'll see that those options are both accepted for the solver). Then the two following lines, 
```
if "." in solver_name:
    _, solver_name = solver_name.split(".")
```
extract the "cbc" or "gurobi" substring. 
However, @ggsdc , since your config schema also allows the value "right_corner" for the solver, you should probably add an else to the if:
```
if "." in solver_name:
    _, solver_name = solver_name.split(".")
else:
    solver_name = "cbc"
```
Or just remove the `right_corner` option from the config schema since it would be the same as `right_solver.cbc` anyway.

Reply:
ok, I didn't know that you could set the solver that way.